### PR TITLE
Feature/audit

### PR DESCRIPTION
**Pull request by dwurf**
## Proposed change
Add the audit library

## Relevant links (github issues, JIRA tickets or ADRs)
https://flip-eng.atlassian.net/browse/GEPPIE-41
http://drive.google.com/open?id=11TgwCKzirUkQR-tcNLHEY48fvynIM3IM4LmsiTofCCE&authuser=0
https://flip-eng.atlassian.net/wiki/spaces/GEPPETTO/pages/100270093/Audit+Service+Design+Notes

## Sensible defaults violation
None identified. Audit logs conform to the [audit event structure](https://flip-eng.atlassian.net/wiki/spaces/GEPPETTO/pages/102924289/Audit+Event+Structure) rather than the generic log structure.

## Associated Metrics
Keep an eye on performance, there may be a latency penalty to using S3 when writing audit logs

## Cost
Use of this library will incur s3 costs at the standard rates for each audit event.

## Additional comments
IAM role will be required

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,4 +2,57 @@
 
 Functions for logging and metrics
 
-Skeleton package for testing packaging
+##Grabbing the library
+
+Put the following in your `requirements.txt`. Change `v0.0.3` to match a
+branch, tag or commit hash.
+
+    git+https://github.com/flipgroup/flippy-logs@v0.0.3#subdirectory=source
```
Comment (dwurf) [inline README.md]:
```suggestion
    git+https://github.com/flipgroup/flippy-logs@v0.0.7#subdirectory=source
```

File: README.md [hunk 2]
```diff
@@ -2,4 +2,57 @@
 
 Functions for logging and metrics
 
-Skeleton package for testing packaging
+##Grabbing the library
+
+Put the following in your `requirements.txt`. Change `v0.0.3` to match a
```
Comment (dwurf) [inline README.md]:
```suggestion
Put the following in your `requirements.txt`. Change `v0.0.7` to match a
```

File: source/flippy_logs/audit.py [hunk 3]
```diff
@@ -0,0 +1,251 @@
+"""Audit library - for logging audit events (user did x)
+
+Audit usage:
+
+    from flippy_logs.audit import Audit
+    # One-off setup of audit logger
+    audit = Audit(
```
Comment (ggershoni) [inline source/flippy_logs/audit.py]: Why not make this a doctest?

File: source/setup.py [hunk 4]
```diff
@@ -2,16 +2,21 @@
...
+"""
+
+VERSION = '0.0.8'
 
 setuptools.setup(
     name="flippy_logs",
-    version="0.0.1",
+    version=VERSION,
     author="Wombat team (Flip)",
     author_email="wombat-ops@flipgroup.com.au",
```
Comment (dwurf) [inline source/setup.py]: Update

File: source/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,3 @@
+python-json-logger==0.1.10
+boto3==1.9.97
+datadog==0.26.0
```
Comment (dwurf) [inline source/requirements.txt]: remove

File: source/flippy_logs/test_metrics.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    metrics.emit('test')
+
+    out, err = capsys.readouterr()
+    line = [l for l in out.split('\n') if l.startswith('MONITORING|')]
+
+    assert line
+    assert line[0].endswith('1|count|flip.metrics.testing.test|#')
+
+
+@mock.patch.dict(os.environ, {
```
Comment (dwurf) [inline source/flippy_logs/test_metrics.py]: remove

**Pull request merged**